src/js/actions/MesosActions.js

```diff
@@ -0,0 +1,48 @@
+var AppDispatcher = require("../AppDispatcher");
+var JSONPUtil = require("../helpers/JSONPUtil");
+var MesosEvents = require("../events/MesosEvents");
+
+var MesosActions = {
+  requestState: function (id, url) {
+    JSONPUtil.request(`${url}/state`).then(
```

Comment:
The `state.json` endpoint works, but it's already [marked as deprecated](https://github.com/apache/mesos/blob/ffbed5ea3bb059ed6dd8830d8a6acc5195ed3683/src/master/master.cpp#L826) and will be removed in 0.26.